buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java

```diff
@@ -103,10 +103,9 @@ public OpenSearchCluster(
         this.nodes = project.container(OpenSearchNode.class);
         this.bwcJdk = bwcJdk;
 
-        this.nodes.add(
-            new OpenSearchNode(path, clusterName + "-0", project, reaper, fileSystemOperations, archiveOperations, workingDirBase, bwcJdk)
-        );
-        // configure the cluster name eagerly so nodes know about it
+        // Always add the first node
+        addNode(clusterName + "-0");
+        // configure the cluster name eagerly so all nodes know about it
         this.nodes.all((node) -> node.defaultConfig.put("cluster.name", safeName(clusterName)));
```

Comment:
This seems redundant now that you are setting cluster name within addNode on line 144.

Reply:
It's needed to initialize the clusterName on any nodes that are configured in the [container itself](https://github.com/kartg/OpenSearch/blob/73ce7bbbe40af252873b5fe99dda9c8b9c938022/buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java#L103), and not initialized by the `addNode` logic